### PR TITLE
Support Platform Access Token

### PR DESCRIPTION
**Pull request by RobiNino**
- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

Breaking Changes in Access Token service:
`CommonTokenParams.ExpiresIn`: changed from `int` to `*uint`. Pointer to allow sending it empty, thus providing the token with the platform's default expiry. `uint` because it does not accept negative values.

Besides the changes above, this PR adds some missing fields to the access token structure.

### CODE REVIEW

File: access/services/accesstoken.go [hunk 1]
```diff
@@ -21,11 +21,14 @@ type TokenService struct {
...
-	IncludeReferenceToken *bool `json:"include_reference_token,omitempty"`
+	IncludeReferenceToken *bool  `json:"include_reference_token,omitempty"`
+	Username              string `json:"username,omitempty"`
+	ProjectKey            string `json:"project_key,omitempty"`
+	Description           string `json:"description,omitempty"`
 }
 
-func NewCreateTokenParams(params CreateTokenParams) CreateTokenParams {
-	return CreateTokenParams{CommonTokenParams: params.CommonTokenParams, IncludeReferenceToken: params.IncludeReferenceToken}
+func NewCreateTokenParams() CreateTokenParams {
```
Comment (yahavi) [inline access/services/accesstoken.go:30]: Can we add this to the readme?

**Review by yahavi — CHANGES_REQUESTED** [review]

File: access/services/accesstoken.go [hunk 2]
```diff
@@ -70,27 +69,29 @@ func (ps *TokenService) createAccessToken(params CreateTokenParams) (auth.Create
...
 }
 
-func (ps *TokenService) addAccessTokenAuthorizationHeader(params CreateTokenParams, httpDetails *httputils.HttpClientDetails) error {
-	access := ps.ServiceDetails.GetAccessToken()
-	if access == "" {
-		access = params.AccessToken
+func (ps *TokenService) handleUnauthenticated(params CreateTokenParams, httpDetails *httputils.HttpClientDetails) error {
+	// Creating access tokens using username and password is available since Artifactory 7.63.2,
+	// by enabling "Enable Token Generation via API" in the UI.
+	if httpDetails.AccessToken != "" || (httpDetails.User != "" && httpDetails.Password != "") {
```
[the remark this answers is not in the document]
Comment (RobiNino) [reply access/services/accesstoken.go:75]: No, ApiKey is not supported by access

File: access/services/accesstoken.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (RobiNino) [reply access/services/accesstoken.go:30]: I think we better of leaving these as they are, and in the future deprecate the all the Artifactory related Access Tokens code.

Comment (RobiNino) [reply access/services/accesstoken.go:30]: Undoing this change

**Pull request merged**